pkg/kubelet/sysctl/safe_sysctls.go

```diff
@@ -51,26 +60,40 @@ var safeSysctlsIncludeReservedPorts = []string{
 // - it is namespaced in the container or the pod
 // - it is isolated, i.e. has no influence on any other pod on the same node.
 func SafeSysctlAllowlist() []string {
-	if goruntime.GOOS == "linux" {
-		// make sure we're on a new enough kernel that the ip_local_reserved_ports sysctl is namespaced
-		kernelVersion, err := getKernelVersion()
-		if err != nil {
-			klog.ErrorS(err, "Failed to get kernel version, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list")
-			return safeSysctls
+	if goruntime.GOOS != "linux" {
+		return nil
+	}
+	return getSafeSysctlAllowlist(getKernelVersion)
+}
+
+func getSafeSysctlAllowlist(getVersion func() (*version.Version, error)) []string {
+	kernelVersion, err := getVersion()
+	if err != nil {
+		klog.ErrorS(err, "failed to get kernel version, unable to determine which sysctls are available")
+	}
+
+	var safeSysctlAllowlist []string
+	for _, sc := range safeSysctls {
+		if sc.kernel == "" {
+			safeSysctlAllowlist = append(safeSysctlAllowlist, sc.name)
+			continue
 		}
-		if kernelVersion.LessThan(version.MustParseGeneric(ipLocalReservedPortsMinNamespacedKernelVersion)) {
-			klog.ErrorS(nil, "Kernel version is too old, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list", "kernelVersion", kernelVersion)
-			return safeSysctls
+
+		if kernelVersion != nil && kernelVersion.AtLeast(version.MustParseGeneric(sc.kernel)) {
+			safeSysctlAllowlist = append(safeSysctlAllowlist, sc.name)
+		} else {
+			klog.ErrorS(nil, "drop the sysctl from safe sysctl list", "sysctl", sc.name, "kernelVersion", kernelVersion)
```

Comment:
Hm, I like the old error message better. Let's merge them together:

```
klog.ErrorS(nil, "Kernel version is too old, dropping sysctl from safe sysctl list", "kernelVersion", kernelVersion, "sysctl", sc.name)
```

Reply:
Actually, This is not necessarily because the kernel version is too small, it may also be because the kernel version cannot be obtained...